docs/examples/use_cases/paddle/ssd/train.py

```diff
@@ -23,94 +23,82 @@
-                                              num_threads,
-                                              device_id,
-                                              seed=42 + device_id)
-        self.reader = ops.COCOReader(
-            file_root=file_root,
-            annotations_file=annotations_file,
-            skip_empty=True,
-            shard_id=local_rank,
-            num_shards=world_size,
-            ratio=True,
-            ltrb=True,
-            shuffle_after_epoch=True,
-            pad_last_batch=True)
-
-        self.crop = ops.RandomBBoxCrop(
-            device="cpu",
-            aspect_ratio=[0.5, 2.0],
-            thresholds=[0, 0.1, 0.3, 0.5, 0.7, 0.9],
-            scaling=[0.3, 1.0],
-            bbox_layout="xyXY",
-            allow_no_crop=True,
-            num_attempts=50)
-        self.bbflip = ops.BbFlip(device="cpu", ltrb=True)
-
-        self.roi_decode = ops.ImageDecoderSlice(device="mixed")
-        self.resize = ops.Resize(
-            device="gpu",
-            resize_x=300,
-            resize_y=300,
-            min_filter=types.DALIInterpType.INTERP_TRIANGULAR)
-        self.hsv = ops.Hsv(device="gpu", dtype=types.FLOAT)  # use float to avoid clipping and
+    pipeline = Pipeline(batch_size, num_threads,
+                        local_rank, seed=42 + device_id)
+
+    with pipeline:
+        images, bboxes, labels = fn.coco_reader(file_root=file_root,
+                                                annotations_file=annotations_file,
+                                                skip_empty=True,
+                                                shard_id=local_rank,
+                                                num_shards=world_size,
+                                                ratio=True,
+                                                ltrb=True,
+                                                random_shuffle=False,
+                                                shuffle_after_epoch=True,
+                                                name="Reader")
+
+        crop_begin, crop_size, bboxes, labels = fn.random_bbox_crop(bboxes, labels,
+                                                                    device="cpu",
+                                                                    aspect_ratio=[
+                                                                        0.5, 2.0],
+                                                                    thresholds=[
+                                                                        0, 0.1, 0.3, 0.5, 0.7, 0.9],
+                                                                    scaling=[
+                                                                        0.3, 1.0],
+                                                                    bbox_layout="xyXY",
+                                                                    allow_no_crop=True,
+                                                                    num_attempts=1)
```

Comment:
Compliant as it may be, the old code was much, _much_, **much** easier to read.

Reply:
Changed a bit